ginac/numeric.cpp

```diff
@@ -1801,8 +1801,10 @@ const numeric numeric::power(signed long exp_si) const
 const numeric numeric::pow_intexp(const numeric &exponent) const
 {
         if (not exponent.is_integer())
-                throw std::runtime_error("nueric::pow_intexp: exponent not integer");
-        if (exponent.t == MPZ) {
+                throw std::runtime_error("numeric::pow_intexp: exponent not integer");
+	// Because Sage may have simplified a fraction to an integer,
+	// we should check whether the type is MPQ as well as MPZ
+        if (exponent.t == MPZ || exponent.t == MPQ) {
                 if (not mpz_fits_sint_p(exponent.v._bigint))
                         throw std::runtime_error("size of exponent exceeds signed long size");
                 return power(mpz_get_si(exponent.v._bigint));
```

Comment:
We should handle MPQ in a separate branch. MPQ values are stored in `_bigrat`. We get lucky with the `mpz_fints_sint_p(exponent.v._bigint)` call below since [`Value` is a union](https://github.com/pynac/pynac/blob/3c109994a50715e4a94e39e2424f693dee065c6b/ginac/numeric.h#L84) and[ `fmpq` is a struct with `num` and `den` members](https://github.com/wbhart/flint2/blob/86d31f6a427c932b4c353576a9d0f561d07045ed/fmpq.h#L39) laid out in that order.

```suggestion
        if (exponent.t == MPZ) {
            if (not mpz_fits_sint_p(exponent.v._bigint))
                throw std::runtime_error("size of exponent exceeds signed long size");
            return power(mpz_get_si(exponent.v._bigint));
        } else if (exponent.t == MPQ) {
            if (not mpz_fits_sint_p(exponent.v._bigrat.num))
                throw std::runtime_error("size of exponent exceeds signed long size");
            return power(mpz_get_si(exponent.v._bigrat.num));
```